source/connection.cc

```diff
@@ -209,6 +211,21 @@ RestClient::Connection::SetKeyPath(const std::string& keyPath) {
   this->keyPath = keyPath;
 }
 
+void
+RestClient::Connection::SetProxy(const std::string& uriProxy) {
+  if (uriProxy.empty())
+    return;
+
+  std::string uriProxyUpper = uriProxy;
+  std::transform(uriProxyUpper.begin(), uriProxyUpper.end(),
+    uriProxyUpper.begin(), ::toupper);
+
+  if (uriProxyUpper.compare(0, 4, "HTTP") != 0)
+    this->uriProxy = "http://" + uriProxy;
+  else
+    this->uriProxy = uriProxy;
```

Comment:
The most important thing is to mention the port of the proxy at the end of the adress, the prefix if the address whether it's http:// or https:// doesn't matter. An HTTPS proxy is a HTTP proxy. More info here : https://curl.haxx.se/libcurl/c/CURLOPT_PROXY.html , here https://curl.haxx.se/libcurl/c/CURLOPT_HTTPPROXYTUNNEL.html and here https://curl.haxx.se/libcurl/c/CURLOPT_PROXYTYPE.html